# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

In [33]:
import pandas as pd
from math import sqrt
import numpy as np
import matplotlib.pyplot as plt
% matplotlib inline
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

## 1. What test is appropriate for this problem? Does CLT apply?

A two sample test for the difference in call back proportions between black sounding and white sounding names is approapriate for this study. Since the sample sizes are sufficiently large, there are np > 10 successes (calls) and nq >  10 failures (no calls) for both black and white sounding names, and the names were randomly applied to resumes, CLT does apply.

## 2. What are the null and alternate hypotheses?

The null hypothesis is that there is no difference in call back proportions between white and black sounding names. The alternative hypothesis is that there is a difference between white and black sounding names. Our alpha threshold at which we will reject the null hypotheses is 0.05.



$$H0: p1 = p2 $$
$$H1: p1 \neq p2 $$
$$a = 0.05 $$

## 3. Compute margin of error, confidence interval, and p-value.

### Margin of error

In [60]:
# Calculate call back proportions
n_black = len(data[data.race == 'b'])
n_white = len(data[data.race == 'w'])
call_black = sum(data.call[data.race == 'b'])
call_white = sum(data.call[data.race == 'w'])
p_black = call_black/n_black
p_white = call_white/n_white

# Calculate difference between proportions
diff = p_white - p_black
# Calculate standard error
SE = sqrt( ((p_white*(1 - p_white))/n_white) + ((p_black*(1 - p_black))/n_black)) 
# Get z value for a 95% confidence level
z_val = stats.norm.ppf(0.975)
# Calculate Margin of Error
ME = z_val*SE
print("The margin of error is %.3f" % ME)

The margin of error is 0.015


### Confidence Interval

In [73]:
# Compute a 95% confidence interval
lower, upper = diff + ME * pd.Series([-1, 1])
print("The lower and upper bounds for a 95-percent confidence interval for the difference between call back proportions are %.4f and %.4f, respectively." % (lower, upper))


The lower and upper bounds for a 95-percent confidence interval for the difference between call back proportions are 0.0168 and 0.0473, respectively.


### P-value

In [80]:
# Compute the p-value for the difference in proportions
diff_test = stats.ttest_ind(data.call[data.race == 'w'], data.call[data.race == 'b'])
print("The probability of getting a magnitude %.3f of difference between call back rates for black and white sounding names on resumes is %.10f." % (diff, diff_test[1]))

The probability of getting a magnitude 0.032 of difference between call back rates for black and white sounding names on resumes is 0.0000394080.


## 4. Write a story describing the statistical significance in the context or the original problem.

As indicated above, the probability of seeing a difference of 0.032 between call back rates between black and white sounding names on resumes is less than our alpha threshold of 0.05, which leads us to reject the null hypothesis that the proportion of call backs is the same between the two samples. Furthermore, we can say with 95% confidence that the true difference between proportions is between approximately 2% and 5% indicating that white sounding names on resumes lead to approximately 2% to 5% higher call back rates than that of their black sounding name counterparts. This may not sound like a practically significant effect size, but when you consider this effect accross 1M candidates, that translates to between 20,000 and 50,000 more white sounding name resumes getting called back. 

## 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The objective of this study was to measure the effect of race/name on resume call back rates with all other factors being held equal. The results of the analysis indicate that there is a statistically significant effect, but further analysis would need to be done to determine which factor is the most important in determining call back success. To measure the effect size of other factors (i.e. education, experience, gender, job match, etc.), it would be necessary to perform a multivariate analysis which allowed us to see the effect of each factor holding all other factors equal.